### PR TITLE
Add toggle to disable automatic seeking to object after placement

### LINKED ISSUE #22668: Add toggle to disable automatic seeking to object after placement
Going to escalate this to an issue since the direction is established.

- Should be a toggle in editor toolbar
- Should apply to all rulesets
- Should default to automatic seek enabled (preserving current behaviour)

### Discussed in https://github.com/ppy/osu/discussions/17424

<div type='discussions-op-text'>

<sup>Originally posted by **641i130** March 23, 2022</sup>
The option to turn off the auto-scroll when writing a beatmap in mania would be very useful. </div>

### PR DESCRIPTION
**Pull request by MaxKruse**
fixes #22668 

Test for mania specifically included in `TestScenePlacementBeforeTrackStart`, unsure if another place would be better for this.

### CODE REVIEW

File: osu.Game/Screens/Edit/Editor.cs [hunk 1]
```diff
@@ -185,6 +185,7 @@ protected override IReadOnlyDependencyContainer CreateChildDependencies(IReadOnl
 
         private Bindable<float> editorBackgroundDim;
         private Bindable<bool> editorHitMarkers;
+        private Bindable<bool> editorSeekToHitobject;
```
Comment (bdach) [inline osu.Game/Screens/Edit/Editor.cs:188]: nit: rename to `editorSeekToHitObject` (goes everywhere else too, I'll spare us both posting the same comment everywhere)

File: osu.Game/Rulesets/Edit/HitObjectComposer.cs [hunk 2]
```diff
@@ -70,6 +71,7 @@ public abstract partial class HitObjectComposer<TObject> : HitObjectComposer, IP
         private FillFlowContainer togglesCollection;
 
         private IBindable<bool> hasTiming;
+        protected Bindable<bool> SeekToHitobject { get; private set; }
```
Comment (bdach) [inline osu.Game/Rulesets/Edit/HitObjectComposer.cs:74]: why `protected`? looks like `private` would suffice

File: osu.Game/Rulesets/Edit/HitObjectComposer.cs [hunk 3]
```diff
@@ -365,6 +369,9 @@ public void EndPlacement(HitObject hitObject, bool commit)
             {
                 EditorBeatmap.Add(hitObject);
 
+                // conditionally seek based on setting
```
Comment (bdach) [inline osu.Game/Rulesets/Edit/HitObjectComposer.cs:372]: remove comment - it restates the code and brings no value

File: osu.Game/Localisation/EditorStrings.cs [hunk 4]
```diff
@@ -19,6 +19,11 @@ public static class EditorStrings
         /// </summary>
         public static LocalisableString ShowHitMarkers => new TranslatableString(getKey(@"show_hit_markers"), @"Show hit markers");
 
+        /// <summary>
+        /// "Seek to Object after placement"
+        /// </summary>
+        public static LocalisableString SeekToHitobject => new TranslatableString(getKey(@"seek_to_hitobject"), @"Seek to Object after placement");
```
Comment (bdach) [inline osu.Game/Localisation/EditorStrings.cs:25]: random capitalisation

```suggestion
        /// <summary>
        /// "Seek to object after placement"
        /// </summary>
        public static LocalisableString SeekToHitObject => new TranslatableString(getKey(@"seek_to_hit_object"), @"Seek to object after placement");
```

\+ update references to match

**Review by bdach — DISMISSED** [review]

File: osu.Game/Screens/Edit/Editor.cs [hunk 5]
```diff
@@ -329,6 +331,10 @@ private void load(OsuConfigManager config)
                                             new ToggleMenuItem(EditorStrings.ShowHitMarkers)
                                             {
                                                 State = { BindTarget = editorHitMarkers },
+                                            },
+                                            new ToggleMenuItem(EditorStrings.SeekToHitObject)
```
Comment (bdach) [inline osu.Game/Screens/Edit/Editor.cs:335]: This is in the "view" menu, but it isn't really a "view" option, so it kind of doesn't belong. Not sure what to do about this. @ppy/team-client your thoughts would be appreciated.

**Review by bdach — COMMENTED** [review]
This needed more cleanup. I've cleaned up as much as I knew how to clean up, but I am going to need to get one UX thing past a second pair of eyes.

**Comment by peppy** [comment]
@bdach is this ready for review?

**Comment by bdach** [comment]
@peppy Yeah I was in the middle of getting through this when you typed the comment :sweat_smile: I think it should be relatively alright except for the one open question from the review above.

**Comment by peppy** [comment]
One thing to note going forward is we will probably add an "auto advance" to osu! editor at some point, ie. one beat forward from actual placement. Which is slightly different to this option but also applied for the same reason.

So one way we could handle this is renaming the option to "Seek after placement" to generalise for that case to. In other words, remove the part which implies this will always seek *to the object specifically*.

**Comment by bdach** [comment]
Sure, but to me that's still not really a "view" option, because it doesn't affect the editor visually like any other option in that menu, it changes its behaviour.

Could just repurpose the "view" menu into "options" and call it a day? Dunno. I don't much care, it can live in "view" for now, I just wanted to raise it as a point I find weird in how things are written right now.

**Comment by peppy** [comment]
I think under "view" is fine for now until we have more menus. I wouldn't be wanting to add new top level menus just for one item yet.

**Comment by bdach** [comment]
Ok, well, if we want to future-proof this for the auto-advance case we may want to swap the `bool` type of the setting for an actual enum. Not sure if it needs to happen right now.

**Comment by peppy** [comment]
Well I was thinking more just change the text before it starts to get localised. I'd still hope it was a bool, but just reads as "Auto-seek on placement" or similar.

**Comment by bdach** [comment]
Fair. Are you going to grab that or shall I?

**Comment by peppy** [comment]
All yours, I think you'll get the wording better than I can anyway 😅

**Comment by bdach** [comment]
I've attempted something, let me know if it meets your expectations.

**Review by bdach — APPROVED** [review]

**Review by peppy — APPROVED** [review]

**Pull request merged**